working/0110-incremental-sound-nnbd/roadmap.md

```diff
@@ -0,0 +1,874 @@
+# Sound non-nullable (by default) types with incremental migration 
+
+Author: leafp@google.com
+
+This document proposes a roadmap for implementing a sound null tracking type
+system in Dart similar to what has been
+explored [previously](https://github.com/dart-lang/sdk/pull/28619/files).  The
+purpose of this document is not to describe the technical details of a proposal,
+but instead to argue for a specific set of high-level goals and design choices
+that define the key properties of the final system, and of the migration path to
+get there.  Specifically, we propose to aim for a system which allows for an
+incremental opt-in migration, and which is fully sound once all code in a
+compilation target has opted-in.
+
+## Motivation for non-nullability in Dart.
+
+This has been fairly well explored previously, so just in brief.  This is one of
+the most requested features in developer surveys, and one of the thing that
+developers most frequently mention missing from other languages.  Kotlin, Swift,
+C#, Typescript, and numerous other languages now have support for
```

Comment:
Isn't C# in the process of adding it for c# 8? Maybe s/now have/now have, or are adding,/ ?

Reply:
done.